shell.c

```diff
@@ -1,11 +1,213 @@
 #include "emulator.h"
 #include <ctype.h>
+
 #include <math.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#define JOYSTICK_DEAD_ZONE 8000
+
+// static SDL_Renderer *renderer = NULL;
+// static SDL_Texture *texture = NULL;
+static SDL_Event e;
+bool has_event = false;
+
+static int speed = 1;
+static bool should_quit = false;
+bool colored_screen;
+
+void
+io_processor(i8080 *cpu) // NOLINT(readability-function-cognitive-complexity)
+{
+
+  has_event = SDL_PollEvent(&e) != 0;
+  if (has_event) // NOLINT
+    {
+      if (e.type == SDL_QUIT)
+        {
+          should_quit = true;
+          exit(1);
```

Comment:
Do we want this to exit with an error code? Is this not the normal exit condition?